pkg/client/client_test.go

```diff
@@ -164,6 +185,40 @@ func TestApplySingle(t *testing.T) {
 	})
 }
 
+func newClient(t *testing.T, s *runtime.Scheme) (*applyCl.ApplyClient, *test.FakeClient) {
+	cli := NewFakeClient(t)
+	cli.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+		// force the Generation to `1` for newly created objects
+		if obj, ok := obj.(*unstructured.Unstructured); ok {
+			obj.SetGeneration(1)
+		}
+		return cli.Client.Create(ctx, obj, opts...)
+	}
+	cli.MockUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+		// compare the specs (only) and only increment the generation if something changed
+		// (the server will check the object metadata, but we're skipping this here)
+		if svc, ok := obj.(*corev1.Service); ok {
+			existing := corev1.Service{}
+			if err := cli.Get(ctx, types.NamespacedName{Namespace: svc.GetNamespace(), Name: svc.GetName()}, &existing); err != nil {
+				return err
+			}
+			if !reflect.DeepEqual(existing.Spec, svc.Spec) { // Service has a `spec` field
```

Comment:
may I have one proposal? could we change this to:
```suggestion
			if !reflect.DeepEqual(existing.Spec, svc.Spec) || !reflect.DeepEqual(existing.ObjectMeta.Labels, svc.ObjectMeta.Labels) { // Service has a `spec` field
```
and add a new test case that is proposed below (or maybe above :-))?

Reply:
what would be the benefit of this test? I just wanted to mimic the server behavior but in a simpler form.

Reply:
> what would be the benefit of this test? 

I wanted to have a way to test case when the `CreateOrUpdateObject` based on the changed generation value

> I just wanted to mimic the server behavior but in a simpler form.

yes, you are right. I thought that the generation is incremented also for changes in labels and annotations, but I found out that it is incremented only for changes in spec. 
So keep it as it is and you can ignore my proposal ;-)